backend/src/server.ts

```diff
@@ -50,12 +50,14 @@ class Server {
   }
 
   async start() {
+    const preferences = Preferences.get()
+    const HOST = preferences.disableLocalNetwork ? 'localhost' : '0.0.0.0'
```

Comment:
```suggestion
    const HOST = preferences.disableLocalNetwork ? IP_PRIVATE : IP_OPEN
```
You can get these from the shared constants file.